base/unicode/utf8proc.jl

```diff
@@ -1,70 +1,29 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 # Various Unicode functionality from the utf8proc library
-module UTF8proc
 
-import Base: show, showcompact, ==, hash, string, symbol, isless, length, eltype, start, next, done, convert, isvalid, lowercase, uppercase
+import Base: ==
 
-export isgraphemebreak
+export isgraphemebreak, GraphemeIterator
 
 # also exported by Base:
-export normalize_string, graphemes, is_assigned_char, charwidth, isvalid,
-   islower, isupper, isalpha, isdigit, isnumber, isalnum,
-   iscntrl, ispunct, isspace, isprint, isgraph, isblank
-
-# whether codepoints are valid Unicode scalar values, i.e. 0-0xd7ff, 0xe000-0x10ffff
-isvalid(::Type{Char}, ch::Unsigned) = !((ch - 0xd800 < 0x800) | (ch > 0x10ffff))
-isvalid(::Type{Char}, ch::Integer) = isvalid(Char, Unsigned(ch))
-isvalid(::Type{Char}, ch::Char) = isvalid(Char, UInt32(ch))
-
-isvalid(ch::Char) = isvalid(Char, ch)
-
-# utf8 category constants
-const UTF8PROC_CATEGORY_CN = 0
-const UTF8PROC_CATEGORY_LU = 1
-const UTF8PROC_CATEGORY_LL = 2
-const UTF8PROC_CATEGORY_LT = 3
-const UTF8PROC_CATEGORY_LM = 4
-const UTF8PROC_CATEGORY_LO = 5
-const UTF8PROC_CATEGORY_MN = 6
-const UTF8PROC_CATEGORY_MC = 7
-const UTF8PROC_CATEGORY_ME = 8
-const UTF8PROC_CATEGORY_ND = 9
-const UTF8PROC_CATEGORY_NL = 10
-const UTF8PROC_CATEGORY_NO = 11
-const UTF8PROC_CATEGORY_PC = 12
-const UTF8PROC_CATEGORY_PD = 13
-const UTF8PROC_CATEGORY_PS = 14
-const UTF8PROC_CATEGORY_PE = 15
-const UTF8PROC_CATEGORY_PI = 16
-const UTF8PROC_CATEGORY_PF = 17
-const UTF8PROC_CATEGORY_PO = 18
-const UTF8PROC_CATEGORY_SM = 19
-const UTF8PROC_CATEGORY_SC = 20
-const UTF8PROC_CATEGORY_SK = 21
-const UTF8PROC_CATEGORY_SO = 22
-const UTF8PROC_CATEGORY_ZS = 23
-const UTF8PROC_CATEGORY_ZL = 24
-const UTF8PROC_CATEGORY_ZP = 25
-const UTF8PROC_CATEGORY_CC = 26
-const UTF8PROC_CATEGORY_CF = 27
-const UTF8PROC_CATEGORY_CS = 28
-const UTF8PROC_CATEGORY_CO = 29
-
-const UTF8PROC_STABLE    = (1<<1)
-const UTF8PROC_COMPAT    = (1<<2)
-const UTF8PROC_COMPOSE   = (1<<3)
-const UTF8PROC_DECOMPOSE = (1<<4)
-const UTF8PROC_IGNORE    = (1<<5)
-const UTF8PROC_REJECTNA  = (1<<6)
-const UTF8PROC_NLF2LS    = (1<<7)
-const UTF8PROC_NLF2PS    = (1<<8)
-const UTF8PROC_NLF2LF    = (UTF8PROC_NLF2LS | UTF8PROC_NLF2PS)
-const UTF8PROC_STRIPCC   = (1<<9)
-const UTF8PROC_CASEFOLD  = (1<<10)
-const UTF8PROC_CHARBOUND = (1<<11)
-const UTF8PROC_LUMP      = (1<<12)
-const UTF8PROC_STRIPMARK = (1<<13)
+export normalize_string, graphemes, charwidth
+
+const STABLE    = (1<<1)
```

Comment:
Well, I think UTF8PROC is misleading - because these not just for UTF8, they are really more related to the things you want to do in general Unicode normalization. My next push puts all of these into a Unicode module, so you'd use Unicode.COMPAT, Unicode.COMPOSE, etc., and they won't conflict anywhere.